### ISSUE TITLE
[REVIEW]: Virtual Bumblebees

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#256]
**Submitting author:** @howardjp (<a href="http://orcid.org/0000-0003-4530-1547">James P. Howard, II</a>)
**Repository:** <a href="https://github.com/howardjp/bumblebees" target ="_blank">https://github.com/howardjp/bumblebees</a>
**Version:** 1.1
**Editor:** @arfon
**Reviewer:** @pjotrp
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.574887" target="_blank">10.5281/zenodo.574887</a>

## Status

[![status](http://joss.theoj.org/papers/d2ee3e52e9cc4d7979ee28f4317c01fd/status.svg)](http://joss.theoj.org/papers/d2ee3e52e9cc4d7979ee28f4317c01fd)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/d2ee3e52e9cc4d7979ee28f4317c01fd"><img src="http://joss.theoj.org/papers/d2ee3e52e9cc4d7979ee28f4317c01fd/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/d2ee3e52e9cc4d7979ee28f4317c01fd/status.svg)](http://joss.theoj.org/papers/d2ee3e52e9cc4d7979ee28f4317c01fd)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/howardjp/bumblebees" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/howardjp/bumblebees">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (1.1)?
- [x] **Authorship:** Has the submitting author (howardjp) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @pjotrp it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by pjotrp** [comment]
@howardjp: I started review. Thank you for your contribution to JOSS. Two minor comments:

1. Please merge your development branch with master
2. Check github recognizes the LICENSE - in the repo header

When that is done I'll tick the final boxes.

**Comment by howardjp** [comment]
1. Check, now version 1.2.1
2. Wow, I just want to say, that was an entirely nontrivial operation

**Comment by pjotrp** [comment]
Your software is compliant ;). One last niggle, can you make a release of the software under version 1.2.1? Once that is done we can in principle assign a DOI.

**Comment by pjotrp** [comment]
Thank you @howardjp. The release should match the paper version. 

@arfon I think we can go ahead and accept this submission.

**Comment by arfon** [comment]
@howardjp - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by howardjp** [comment]
Awesome! http://doi.org/10.5281/zenodo.574887

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.574887 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.574887" target="_blank">10.5281/zenodo.574887</a> is the archive.

**Comment by arfon** [comment]
Many thanks for your review @pjotrp ✨ 

@howardjp - your paper is now accepted into JOSS and your paper DOI is http://dx.doi.org/10.21105/joss.00256 ⚡️ 🚀 💥

**Comment by howardjp** [comment]
Thank you @arfon and @pjotrp for seeing this through!
